### PR TITLE
Scoring Udpate

### PR DESCRIPTION
**Pull request by dylanuys**
Addressing incentive volatility via
- New reward function (AUC**2)
- Longer EMA alpha (0.01

### CODE REVIEW

File: bitmind/validator/miner_performance_tracker.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, matthews_corrcoef, roc_auc_score
```
Comment (benliang99) [inline bitmind/validator/miner_performance_tracker.py:1]: Should this be in the `validator/` directory, or somewhere else? Maybe a directory dedicated to miners?

Edit: Nevermind, I misunderstood the usage of this

File: bitmind/validator/reward.py [hunk 2]
```diff
@@ -21,46 +21,53 @@
...
-            reward = 1. if np.round(pred) == label else 0.
-            reward *= count_penalty(pred)
+            miner_hotkey = axon.hotkey
+            if uid in performance_tracker.miner_hotkeys and performance_tracker.miner_hotkeys[uid] != miner_hotkey:
+                bt.logging.info(f"Miner hotkey changed for UID {uid}. Resetting performance metrics.")
+                performance_tracker.reset_miner_history(uid, miner_hotkey)
+
+            performance_tracker.update(uid, pred_prob, label, miner_hotkey)
+            metrics = performance_tracker.get_metrics(uid, window=100)
+            reward = metrics['auc'] ** 2
```
Comment (benliang99) [inline bitmind/validator/reward.py:62]: @dylanuys Suggestion: I think a README with GIFs/figures of the incentive curve experiments could be useful.

**Review by benliang99 — COMMENTED** [review]
LGTM - any specific miner/validator tests we should run before we merge?
